src/components/highchart.js

```diff
@@ -0,0 +1,161 @@
+    }),
+    series: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string,
+        data: PropTypes.arrayOf(
+          PropTypes.shape({
+            y: PropTypes.number
+          })
+        )
+      })
+    ).isRequired,
+    title: PropTypes.string,
+    type: PropTypes.string,
+    xAxis: PropTypes.shape({
+      title: PropTypes.string
+    }),
+    yAxis: PropTypes.shape({
+      title: PropTypes.string
+    })
+  }
+  static defaultProps = {
+    backgroundColor: null,
+    donut: null,
+    legend: null,
+    yAxis: null,
+    xAxis: null,
+    title: null,
+    type: null
+  }
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      config: {}
+    };
+  }
+  componentDidMount() {
+    this.afterRender();
+  }
+  afterRender() {
+    this.config = this.setConfig();
+    if(this.props.type === "radar") {
+      this.configForRadar();
+    } else if(this.props.type === "pie") {
+      this.configForPie();
+    }
+    if(this.props.chart) { this.configForChart(); }
+    if(this.props.xAxis) { this.configForX(); }
+    if(this.props.yAxis) { this.configForY(); }
+    this.configForLegend();
+    this.configForTooltip();
+    this.setState({config: this.config});
+  }
+  setConfig() {
+    return {
+      chart: {
+        type: this.props.type,
+        backgroundColor: this.props.backgroundColor
+      },
+      title: {text: this.props.title},
+      xAxis: {title: {text: ""}, crosshair: true},
+      yAxis: {title: {text: ""}},
+      plotOptions: {
+        column: {
+          pointPadding: 0.2,
+          borderWidth: 0
+        }
+      },
+      credits: {enabled: false},
+      exporting: {enabled: false},
+      legend: {enabled: false},
+      series: this.props.series
+    };
+  }
+  configForRadar() {
+    this.config.chart.type = "area";
+    this.config.chart.polar = true;
+  }
+  configForPie() {
+    this.config.chart = {
+      type: "pie",
+      plotBackgroundColor: null,
+      plotBorderWidth: null,
+      plotShadow: false,
+      height: 270,
+      spacing: [0, 10, 10, 0]
+    };
+    this.config.plotOptions.pie = {
+      allowPointSelect: true,
+      cursor: "pointer",
+      dataLabels: {
+        enabled: true,
+        distance: -30,
+        format: "<b>{point.percentage:.0f}%</b>",
+        style: {
+          color: (HighCharts.theme && HighCharts.theme.contrastTextColor) || "white",
+          textShadow: "0 0 4px #000000"
+        }
+      }
+    };
+    if(this.props.donut) { this.config.series[0].innerSize = `${100 - this.props.donut}%`; }
+    this.config.series[0].colorByPoint = true;
+  }
+  configForX() {
+    Object.assign(this.config.xAxis, this.props.xAxis);
+  }
+  configForY() {
+    Object.assign(this.config.yAxis, this.props.yAxis);
+  }
+  configForChart() {
+    Object.assign(this.config.chart, this.props.chart);
+  }
+  configForLegend() {
+    if(this.props.legend instanceof Object) { this.config.legend = this.props.legend; }
```

Comment:
It looks like `props.legend` can only be an object or null, so I think a simpler check would just be `if(this.props.legend)` like you did for the chart and the x and y axises

Reply:
Might make sense to move the check to the same place as the other ones too